kratos/linear_solvers/amgcl_ns_solver.h

```diff
@@ -240,16 +197,16 @@ class AMGCL_NS_Solver : public LinearSolver< TSparseSpaceType,
 
         amgcl::make_solver<
             amgcl::preconditioner::schur_pressure_correction<USolver, PSolver>,
-            amgcl::runtime::solver::wrapper<Backend>
-            > solve(amgcl::adapter::zero_copy(rA.size1(), rA.index1_data().begin(), rA.index2_data().begin(), rA.value_data().begin()), mprm);
+            amgcl::runtime::solver::wrapper<SolverBackend>
+            > solve(*amgcl::adapter::zero_copy(rA.size1(), rA.index1_data().begin(), rA.index2_data().begin(), rA.value_data().begin()), mprm);
 
         size_t iters;
         double resid;
         std::tie(iters, resid) = solve(rB, rX);
```

Comment:
The solver above uses mixed precision (double precision for the iterative solver, single precision for the preconditioner). The double precision matrix should be passed to the solver here so that the overall solution is double precision:
```suggestion
        std::tie(iters, resid) = solve(*A, rB, rX);
```